keras_maskrcnn/bin/evaluate.py

```diff
@@ -0,0 +1,145 @@
+            args.classes,
+            image_min_side=args.image_min_side,
+            image_max_side=args.image_max_side
+        )
+    else:
+        raise ValueError('Invalid data type received: {}'.format(args.dataset_type))
+
+    return validation_generator
+
+
+def parse_args(args):
+    parser     = argparse.ArgumentParser(description='Evaluation script for a MaskRCNN network.')
+    subparsers = parser.add_subparsers(help='Arguments for specific dataset types.', dest='dataset_type')
+    subparsers.required = True
+
+    coco_parser = subparsers.add_parser('coco')
+    coco_parser.add_argument('coco_path', help='Path to dataset directory (ie. /tmp/COCO).')
+
+    csv_parser = subparsers.add_parser('csv')
+    csv_parser.add_argument('annotations', help='Path to CSV file containing annotations for evaluation.')
+    csv_parser.add_argument('classes', help='Path to a CSV file containing class label mapping.')
+
+    parser.add_argument('model',                help='Path to MaskRCNN model.')
+    parser.add_argument('--backbone',           help='The backbone of the model.', default='resnet50')
+    parser.add_argument('--gpu',                help='Id of the GPU to use (as reported by nvidia-smi).')
+    parser.add_argument('--score-threshold',    help='Threshold on score to filter detections with (defaults to 0.05).', default=0.05, type=float)
+    parser.add_argument('--iou-threshold',      help='IoU Threshold to count for a positive detection (defaults to 0.5).', default=0.5, type=float)
+    parser.add_argument('--max-detections',     help='Max Detections per image (defaults to 100).', default=100, type=int)
+    parser.add_argument('--binarize-threshold', help='Threshold used for binarizing masks (defaults to 0.5).', default=0.5, type=float)
+    parser.add_argument('--save-path',          help='Path for saving images with detections.')
+    parser.add_argument('--image-min-side',     help='Rescale the image so the smallest side is min_side.', type=int, default=800)
+    parser.add_argument('--image-max-side',     help='Rescale the image if the largest side is larger than max_side.', type=int, default=1333)
+
+    return parser.parse_args(args)
+
+
+def main(args=None):
+    # parse arguments
+    if args is None:
+        args = sys.argv[1:]
+    args = parse_args(args)
+
+    # make sure keras is the minimum required version
+    check_keras_version()
+
+    # optionally choose specific GPU
+    if args.gpu:
+        os.environ['CUDA_VISIBLE_DEVICES'] = args.gpu
+    keras.backend.tensorflow_backend.set_session(get_session())
+
+    # make save path if it doesn't exist
+    if args.save_path is not None and not os.path.exists(args.save_path):
+        os.makedirs(args.save_path)
+
+    # create the generator
+    generator = create_generator(args)
+
+    # load the model
+    print('Loading model, this may take a second...')
+    model = models.load_model(args.model, backbone_name=args.backbone)
+
+    # print model summary
```

Comment:
Maybe it might be useful to add a verbose option that also print this

Reply:
Hmm I have doubts about it. When do you think it'd be useful?

Reply:
When debbugging changes to the network (for example what happened to @enricoliscio when workign on the regression), but it's a very uncommon case

Reply:
Hmm, I propose to add that when it comes up.